### DISCUSSION TITLE
Allow parcel to use my custom createElement function instead of React's createElement, while transpiling the code

### DISCUSSION THREAD

**Discussion by abhinav-anshul** [parcel-bundler/parcel#8457] [Q&A]
Hi,

I'm using my own custom createElement function and hitting it while using JSX code. I want Parcel to use my createElement API/function while transpilation

My custom API looks like this :

```
function createElement(type, props, ...children) {
  return {
    type,
    props: {
      ...props,
      children: children.map((child) =>
        typeof child === "object" ? child : createTextElement(child)
      ),
    },
  }
}
```

I'm trying to use this function in my JSX code as :

```
// import { createElement } from "react" <-- NOT THIS
import { createElement } from "./src/schnell"

const App = createElement(
  "h1",
  { id: "headingTitle" },
  createElement("a", null, "bar"),
  createElement("b")
)
```

I'm getting errors as `Error: Objects are not valid as a React child (found: object with keys {type, props}). If you meant to render a collection of children, use an array instead.`

I believe, the JSX transpiler is still using React's API, not my `createElement`? 

This problem is solved in **_babel_** by using

 ```
 const myLibraryName = {
  createElement,
}
 /** @jsx myLibraryName.createElement */
 
 ```
 
 at the top of the file.
 
 How can I achieve the same here?
 
 Thanks for reading!

**Comment by mischnic** [comment]
That `@jsx` comment should also be working with swc. Have you tried that already?
